### ISSUE TITLE
Add mandatory fields to Quick create modals

### ISSUE THREAD

**Issue by AbhishekPAnil** [culturecreates/footlight-app#1445]
Currently, the quick create modals (Place/Person/Organization) include a fixed set of predefined fields, while additional fields are available in the full edit forms for each entity. When a user sets a field as mandatory, it is automatically reflected in the edit form, but not in the quick create modals. Mandatory fields should automatically appear in the quick create modals as well.

**Expected behaviour:** All mandatory fields will appear in associated Quick Create modal and are marked as required (red asterisk). If the user doesn't fill the mandatory field, they cannot save (get the error and notification to add the missing data). When all mandatory fields are filled, they can.

**Comment by sahalali** [comment]
@troughc Are we restring users from creating/updating place/person/organization without any mandatory fields?

**Comment by troughc** [comment]
Yes.

**Comment by AbhishekPAnil** [comment]
The frontend changes for these ticket will be included in #1446 .

**Comment by AbhishekPAnil** [comment]
Frotend tasks :24
Add each fields to the quick create modals if its mandatory:8+8+8

**Comment by sahalali** [comment]
Backend estimate: 12 Hrs
Add validations to all place, person and organization Add-Update APIs.

**Comment by troughc** [comment]
@sahalali I am assigning to you for assessment/assignment.

**Comment by sahalali** [comment]
Front-end changes are yet to start.

**Comment by troughc** [comment]
For @troughc 2 questions:
- how to handle address fields that are made mandatory (example street address)
- do we keep non -mandatory fields in quick create? (for example, in Places - Region, Place type)
cc @AbhishekPAnil

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil 
Fields extra other than required fields in :
1.Place

- Place type
- Region

2.Person

- Occupation
- website

3.Organization

- website
- logo

**Comment by AbhishekPAnil** [comment]
>how to handle address fields that are made mandatory (example street address)
do we keep non -mandatory fields in quick create? (for example, in Places - Region, Place type)
cc @AbhishekPAnil

1.The required address fields will be shown in the quick create modal and rest of them will be automatically filled (if it has a value ) once the user selects the desired address (this fields will not be visible to the user ).
2.We will be showing only the required fields in the quick create modals.

**Comment by troughc** [comment]
Question: what happens when you change a non-mandatory field into a mandatory field, and it has not been filled in an existing entity? 
For example, you make Region mandatory. What happens with the Place that is missing a region? What happens to an event that has a place that is missing a region?

### Educated guess - please check
Place: the next time you go into the Place, you are asked to add the missing field in order to save. The field is highlighted and you cannot save until you fill it. You can exit without saving.

Event with mandatory field: same as above.

Event with a nested entity missing a mandatory field: do you get a warning that mandatory field is missing (hopefully you are told where it is missing from, i.e. the place used in the location)?

**Comment by troughc** [comment]
> > how to handle address fields that are made mandatory (example street address)
> > do we keep non -mandatory fields in quick create? (for example, in Places - Region, Place type)
> > cc [@AbhishekPAnil](https://github.com/AbhishekPAnil)

Thanks for writing this up @AbhishekPAnil - I made a little change for clarity.

1. The required address fields will be shown in the quick create modal **so they can be filled manually by the user or get automatically filled when the search for the address**. 
2. The non-mandatory address fields are automatically filled if the user uses the search field and there is a value.  These non-mandatory fields are not displayed in the quick create modal.  
3. We will be showing only the required fields in the quick create modals (no non-mandatory ones).

**Comment by AbhishekPAnil** [comment]
>Educated guess - please check
Place: the next time you go into the Place, you are asked to add the missing field in order to save. The field is highlighted and you cannot save until you fill it. You can exit without saving.

Event with mandatory field: same as above.

I can confirm that this the flow that is happening right now.

**Comment by troughc** [comment]
@AbhishekPAnil following up on a conversation I just had with @sahalali about aggretators importing events with places or other nested entities that do NOT have all mandatory fields filled (for example, a place that does not have the region).
It is important the aggregators can still import these events as draft (and save nested entities in the database)so that the Admin or Editor can review, fix any issues before publishing - including any missing mandatory fields from nested entities. 

Is this how it works today?

**Comment by sahalali** [comment]
The current system does not allow the creation of places, organizations, or people if any of the mandatory fields are missing. This restriction is enforced by the CMS API and applies to entity creation and updates through Aggregator, CMS UI, and other platforms. This is true even when a user is importing a place from Artsdata or any other external source while adding or modifying an event using "Import from Artsdata" option.

Question (This is in addition to what we discussed on Tuesday)
Regarding user restrictions, should we prevent users from adding a new place from Artsdata if it is missing a mandatory field while they are editing an event? What should the expected behavior be in such cases?

**Comment by AbhishekPAnil** [comment]
@troughc
If a user manually adds an entity by searching through "Imports from Artsdata," they can only save the entity if all required conditions are met. The same applies to nested entities. For example, when selecting a place via "Imports from Artsdata" while creating an event, an error will be thrown if the place lacks the necessary fields.

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil Assign it to Caitlin once the testing is complete for https://github.com/culturecreates/footlight-calendar-api/issues/1338

**Comment by AbhishekPAnil** [comment]
@troughc Could you please share the error message that should be displayed when a user attempts to access the full details page without filling in the required fields in the quick create form?

**Comment by troughc** [comment]
@AbhishekPAnil two things:

First, I noticed in the quick create modal that mandatory fields don't have a star (asterisk) next to them:
In this example, both logo and website are mandatory in mandatory settings (but there is no asterisk) - can you please fix?

<img width="522" alt="Image" src="https://github.com/user-attachments/assets/f55a8c95-8a84-4b63-9e10-3a64c9ec44a5" />



Second, here is the message in English and French that should appear when you click add full details or the Create button in the quick create modal: 

English: Fill in the missing information.
French: Remplir les informations manquantes.

**_And then highlight the mandatory fields._**

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil 
>First, I noticed in the quick create modal that mandatory fields don't have a star (asterisk) next to them:
In this example, both logo and website are mandatory in mandatory settings (but there is no asterisk) - can you please fix?

Please separate it out into a new issue.

**Comment by troughc** [comment]
Thanks @AbhishekPAnil. Let me know when I should do a final round of testing.

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil Assign this to Caitlin once [#1338](https://github.com/culturecreates/footlight-calendar-api/issues/1338) is up in production.

**Comment by troughc** [comment]
@AbhishekPAnil just checking for an update on this issue and https://github.com/culturecreates/footlight-calendar-api/issues/1338

**Comment by troughc** [comment]
Test 1 in **Tout Culture calendar**
adding a[ new event](https://pr-1740.dla06mbe0wyl7.amplifyapp.com/dashboard/6308ef4a7f771f00431d939a/events/6814ed4d6168600064816fe4) and [new place](https://pr-1740.dla06mbe0wyl7.amplifyapp.com/dashboard/6308ef4a7f771f00431d939a/places/6814ed316168600064816fce)
- all mandatory fields in the quick create displayed: PASSED
- required to fill them in before saving: PASSED
-  when I went to look at the new place, one of the mandatory fields was missing... Screenshot:

<img width="868" alt="Image" src="https://github.com/user-attachments/assets/b12892a7-1b8d-4112-924a-1d7f1b327264" />

This doesn't seem to happen with my test cases when I am adding an organizaiton or person. It just happened with Place. 
And it only seems to be an issue when I set "Place type" to be a mandatory field, which is a STANDARD field (not custom like the other ones I chose).
I tested again in [another new event](https://pr-1740.dla06mbe0wyl7.amplifyapp.com/dashboard/6308ef4a7f771f00431d939a/events/6814f080616860006481769f) after making "Territory-Place" (another standard taxonomy) mandatory in [this event](https://pr-1740.dla06mbe0wyl7.amplifyapp.com/dashboard/6308ef4a7f771f00431d939a/places/add-place?id=6814f07e6168600064817689). Territory-Place added properly.
Maybe the problem is only with "Place type" 
Steps to repeat:
1. create a new event
2. quick create a new place in the event
3. click add more details right away. The event is saved. But in the place, one of the mandatory fields is empty
4. Alternative: save in quick create, save the event if you want, then go look at the place. The first mandatory field is missing.

**Comment by troughc** [comment]
Tested: PASSED with Place type.
